com.civicrm.hrjobroles/js/src/job-roles/controllers/hr-job-roles-controller.js

```diff
@@ -1233,17 +1232,16 @@ define([
         return (errorsCount === 0);
       }
 
-      // pubSub Events
-
+      // PubSub Events
       // Triggers when an new contract is created for a contact.
       pubSub.subscribe('contract:created', function(contactId){
         contractIdsFromContact(contactId);
       });
 
-      // Triggers when a contract is deleted for a contact.
-      pubSub.subscribe('contract:deleted', function(data){
-        contractIdsFromContact(data.contactId);
-        jobRolesFromContracts(data.contractId);
+      pubSub.subscribe('contract:deleted', function(data) {
+        contractIdsFromContact(data.contactId).then(function(contractIds) {
+          jobRolesFromContracts(contractIds);
```

Comment:
do we need to check if contractIds.length exists before calling `jobRolesFromContracts`